File: api/core/src/main/kotlin/domain/Status.kt
package domain

import base.Encrypted
import base.Id
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class Status(
    override var _id: Id<Status> = Id(),
    var naloga_id: Id<Naloga>,
    var test_id: Id<Test>,
    var oseba_id: Id<Oseba>,
    var tip: Tip,
    @Contextual val pojasnilo: Encrypted
) : Entiteta<Status> {
    enum class Tip { NEZACETO, NERESENO, NAPACNO, PRAVILNO }
}

File: app/src/app/core/services/api/models/status.ts
import { Id } from './id'
import { Naloga } from './naloga'
import { Oseba } from './oseba'
import { Test } from './test'
export interface Status {
	'_id': Id<Status>
	naloga_id: Id<Naloga>
	oseba_id: Id<Oseba>
	pojasnilo: string
	test_id: Id<Test>
	tip: 'NEZACETO' | 'NERESENO' | 'NAPACNO' | 'PRAVILNO'
}